tests/unittests/portadmin/portadmin_test.py

```diff
@@ -1,152 +1,129 @@
-from nav.portadmin.vlan import FantasyVlan
-
-
-class PortadminResponseTest(unittest.TestCase):
-    def setUp(self):
-        self.profile = Mock()
-        self.profile.snmp_version = 2
-        self.profile.snmp_community = "public"
-
-        self.hpVendor = Mock()
-        self.hpVendor.id = u'hp'
-
-        self.ciscoVendor = Mock()
-        self.ciscoVendor.id = u'cisco'
-
-        self.hpType = Mock()
-        self.hpType.vendor = self.hpVendor
-        self.hpType.get_enterprise_id.return_value = VENDOR_ID_HEWLETT_PACKARD
-
-        self.ciscoType = Mock()
-        self.ciscoType.vendor = self.ciscoVendor
-        self.ciscoType.get_enterprise_id.return_value = VENDOR_ID_CISCOSYSTEMS
-
-        self.netboxHP = Mock()
-        self.netboxHP.type = self.hpType
-        self.netboxHP.ip = '10.240.160.39'
-        self.netboxHP.get_preferred_snmp_management_profile.return_value = self.profile
-
-        self.netboxCisco = Mock()
-        self.netboxCisco.type = self.ciscoType
-        self.netboxCisco.ip = '10.240.160.38'
-        self.netboxCisco.get_preferred_snmp_management_profile.return_value = (
-            self.profile
-        )
-
-        self.snmpReadOnlyHandler = None
-        self.handler = None
-
-    def tearDown(self):
-        self.hpVendor = None
-        self.ciscoVendor = None
-        self.hpType = None
-        self.ciscoType = None
-        self.netboxHP = None
-        self.netboxCisco = None
-        self.handler = None
-        self.snmpReadOnlyHandler = None
-        self.snmpReadWriteHandler = None
-
-    ####################################################################
-    #  HP-netbox
-
-    def test_management_factory_get_hp(self):
-        self.handler = ManagementFactory.get_instance(self.netboxHP)
-        self.assertNotEqual(self.handler, None, 'Could not get handler-object')
-        self.assertIsInstance(self.handler, HP, msg='Wrong handler-type')
-
-    def test_get_vlan_hp(self):
-        self.handler = ManagementFactory.get_instance(self.netboxHP)
+from nav.portadmin.management import ManagementFactory
+from nav.portadmin.snmp.hp import HP
+from nav.portadmin.snmp.cisco import Cisco
+
+
+class TestPortadminManagementFactory:
+    def test_get_hp(self, netbox_hp):
+        handler = ManagementFactory.get_instance(netbox_hp)
+        assert handler != None, "Could not get handler-object"
+        assert isinstance(handler, HP), "Wrong handler-type"
+
+    def test_get_cisco(self, netbox_cisco):
+        handler = ManagementFactory.get_instance(netbox_cisco)
+        assert handler != None, "Could not get handler-object"
+        assert isinstance(handler, Cisco), "Wrong handler-type"
+
+
+class TestPortadminResponseHP:
+    def test_get_vlan_hp(self, handler_hp):
         # get hold of the read-only Snmp-object
-        self.snmpReadOnlyHandler = self.handler._get_read_only_handle()
+        snmpReadOnlyHandler = handler_hp._get_read_only_handle()
```

Comment:
While you're at it, why not clean up the ridiculous camel-casing. PEP-8 for the win! :)

Reply:
Funny part is i fixed camelcase for all other variables except that one :facepalm:

Reply:
done